### PR TITLE
stormlib: 9.22 -> 9.23

### PR DESCRIPTION
**Pull request by AndersonTorres**
## Description of changes

Closes https://github.com/NixOS/nixpkgs/pull/147148

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/st/stormlib/package.nix
Comment (viraptor) [inline pkgs/by-name/st/stormlib/package.nix:2]: I think the common format is still:
```
{ lib
, bzip2
, ...
}
```

File: pkgs/by-name/st/stormlib/package.nix
Comment (viraptor) [inline pkgs/by-name/st/stormlib/package.nix:37]: Did you mean to drop the main program?

File: pkgs/by-name/st/stormlib/package.nix
Comment (eclairevoyant) [reply pkgs/by-name/st/stormlib/package.nix:2]: nixfmt has introduced a new style as per rfc 166.

File: pkgs/by-name/st/stormlib/package.nix
Comment (AndersonTorres) [reply pkgs/by-name/st/stormlib/package.nix:37]: Yes. When I built it, there was no bin/ subdir.

**Comment by aanderse** [comment]
I'll try to test this soon and let you know how it goes

**Review by aanderse — APPROVED** [review]
did some testing - all good, thanks

everyone ok with a merge?

**Comment by eclairevoyant** [comment]
According to borgo, seems that [smpq cannot find stormlib headers on darwin](https://logs.ofborg.org/?key=nixos/nixpkgs.305444&attempt_id=4a368f24-1cb9-44d1-a0fa-100da8ece190), despite the explicit override in [`all-packages.nix`](https://github.com/NixOS/nixpkgs/pull/305444/files#diff-ab5748dc9567516fefba8344056b51ec1866adeace380f46e58a7af3d619ea22)

**Comment by AndersonTorres** [comment]
Was it working before?

**Comment by eclairevoyant** [comment]
Ah guess not, it's been failing since [at least Feb 2023](https://hydra.nixos.org/build/209950144). Can we mark smpq broken on darwin? I'm fine to merge after.

**Comment by viraptor** [comment]
I'm not sure if this has real impact, but raising just in case: When I try building smpq, the stormlib has a mismatch between version and the directory names: (9.22 vs 9.21)
```
Install the project...
-- Install configuration: "Release"
-- Installing: /nix/store/rfmbh1vc8l536wbvlf2daq050r5xjd91-stormlib-9.22/Library/Frameworks/storm.framework
-- Installing: /nix/store/rfmbh1vc8l536wbvlf2daq050r5xjd91-stormlib-9.22/Library/Frameworks/storm.framework/Resources
-- Installing: /nix/store/rfmbh1vc8l536wbvlf2daq050r5xjd91-stormlib-9.22/Library/Frameworks/storm.framework/Versions
-- Installing: /nix/store/rfmbh1vc8l536wbvlf2daq050r5xjd91-stormlib-9.22/Library/Frameworks/storm.framework/Versions/9.21.0
-- Installing: /nix/store/rfmbh1vc8l536wbvlf2daq050r5xjd91-stormlib-9.22/Library/Frameworks/storm.framework/Versions/9.21.0/Resources
...
```

**Comment by eclairevoyant** [comment]
That's helpful info, but I'd rather not block this PR which isn't introducing any regressions, i.e. it's a strict improvement over the current state. In the interest of keeping things moving, I'd lean towards marking it broken and solving the darwin issue in another PR.

**Comment by AndersonTorres** [comment]
Fortunately stormlib affects only smpq.

**Review by eclairevoyant — APPROVED** [review]

**Pull request merged**
